hnn_core/parallel_backends.py

```diff
@@ -230,20 +243,43 @@ def _read_data(self, fd, mask):
 
         # only _read_stdout gets a signal from the fd. the end of simulation
         # data is signalled by the process terminating
-        return False
+        return None
 
     def _read_stdout(self, fd, mask):
         """read from fd until receiving the process simulation is complete"""
         data = os.read(fd, 4096)
         if data:
             str_data = data.decode()
-            if str_data == 'sim_complete':
-                return True
+            if str_data == 'end_of_sim' or str_data.startswith('end_of_data'):
+                return str_data
 
             # output from process includes newlines
             sys.stdout.write(str_data)
 
-        return False
+        return None
+
+    def _process_child_data(self, data_bytes, data_length):
```

Comment:
```suggestion
    def _process_child_data(self, data_bytes, data_len):
```

to keep it consistent